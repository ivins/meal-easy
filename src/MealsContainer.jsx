import React from 'react';
import { TabContent, TabPane, Nav, NavItem, NavLink, Card, Button, CardTitle, CardText, Row, Col } from 'reactstrap';
import classnames from 'classnames';
import { Droppable } from 'react-beautiful-dnd';

export default class Example extends React.Component {
  constructor(props) {
    super(props);

    this.toggle = this.toggle.bind(this);
    this.state = {
      activeTab: '1'
    };
  }

  toggle(tab) {
    if (this.state.activeTab !== tab) {
      this.setState({
        activeTab: tab
      });
    }
  }
  render() {
    return (
      <div className="columnContainer">
        <Nav tabs>
          <NavItem>
            <NavLink
              className={classnames({ active: this.state.activeTab === '1' })}
              onClick={() => { this.toggle('1'); }}
            >
              This Weeks Meal Plan
            </NavLink>
          </NavItem>
          {/* <NavItem>
            <NavLink
              className={classnames({ active: this.state.activeTab === '2' })}
              onClick={() => { this.toggle('2'); }}
            >
              This Week
            </NavLink>
          </NavItem> */}
        </Nav>
        <TabContent activeTab={this.state.activeTab}>
          {/* <TabPane tabId="1">
            <Row>
              <Col sm="12">
                <h4>Tab 1 Contents</h4>
              </Col>
            </Row>
          </TabPane> */}
          <TabPane tabId="1">
          {/* <Droppable> */}
            <Row>
              <Col sm="11" className="mealCard">
                <Card body>
                  <CardTitle>Special Title Treatment</CardTitle>
                  <CardText>With supporting text below as a natural lead-in to additional content.</CardText>
                  <Button>Go somewhere</Button>
                </Card>
              </Col>
              <Col sm="11" className="mealCard">
                <Card body>
                  <CardTitle>Special Title Treatment</CardTitle>
                  {/* <img src="https://spoonacular.com/recipeImages/537176-240x150.jpg" alt="food"/> */}
                  <CardText>With supporting text below as a natural lead-in to additional content.</CardText>
                  <Button>Go somewhere</Button>
                </Card>
              </Col>
              <Col sm="11" className="mealCard">
                <Card body>
                  <CardTitle>Special Title Treatment</CardTitle>
                  {/* <img src="https://spoonacular.com/recipeImages/537176-240x150.jpg" alt="food"/> */}
                  <CardText>With supporting text below as a natural lead-in to additional content.</CardText>
                  <Button>Go somewhere</Button>
                </Card>
              </Col>
            </Row>
            {/* </Droppable> */}
          </TabPane>
        </TabContent>
      </div>
    );
  }
}