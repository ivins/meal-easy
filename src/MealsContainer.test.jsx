import React from 'react';
import ReactDOM from 'react-dom';
import MealsContainer from './MealsContainer'
import Adapter from 'enzyme-adapter-react-16'
import Enzyme, { shallow, render, mount } from 'enzyme'
Enzyme.configure({ adapter: new Adapter() });

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<MealsContainer ideas={[2]}/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('construction of Meals Container', ()=>{
  let props;
  beforeEach( ()=> {
    props = {
      favourites: [{id:1,title:"Food"}],
      ideas: [{id:1,title:"Food"}]
    }
  })

  test('with props renders MealsContainer Component', ()=>{
    const wrapper = shallow(<MealsContainer {...props} />);
    expect(wrapper).toMatchSnapshot();
  })

  test('with no props renders MealsContainer Component', ()=>{
    const wrapper = shallow(<MealsContainer />);
    expect(wrapper).toMatchSnapshot();
  })

})
