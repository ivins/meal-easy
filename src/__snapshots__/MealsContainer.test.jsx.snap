// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`construction of Meals Container with no props renders MealsContainer Component 1`] = `
<div
  className="columnContainer"
>
  <Nav
    tabs={true}
    tag="ul"
    vertical={false}
  >
    <NavItem
      tag="li"
    >
      <NavLink
        className="active"
        onClick={[Function]}
        tag="a"
      >
        <b>
          My Favourites
        </b>
      </NavLink>
    </NavItem>
    <NavItem
      tag="li"
    >
      <NavLink
        className=""
        onClick={[Function]}
        tag="a"
      >
        <b>
          Need Ideas?
        </b>
      </NavLink>
    </NavItem>
  </Nav>
  <TabContent
    activeTab="1"
    tag="div"
  >
    <TabPane
      tabId="1"
      tag="div"
    >
      <Row
        tag="div"
      />
    </TabPane>
    <TabPane
      tabId="2"
      tag="div"
    >
      <Row
        tag="div"
      >
        <SearchInstruction />
      </Row>
    </TabPane>
  </TabContent>
</div>
`;

exports[`construction of Meals Container with props renders MealsContainer Component 1`] = `
<div
  className="columnContainer"
>
  <Nav
    tabs={true}
    tag="ul"
    vertical={false}
  >
    <NavItem
      tag="li"
    >
      <NavLink
        className="active"
        onClick={[Function]}
        tag="a"
      >
        <b>
          My Favourites
        </b>
      </NavLink>
    </NavItem>
    <NavItem
      tag="li"
    >
      <NavLink
        className=""
        onClick={[Function]}
        tag="a"
      >
        <b>
          Need Ideas?
        </b>
      </NavLink>
    </NavItem>
  </Nav>
  <TabContent
    activeTab="1"
    tag="div"
  >
    <TabPane
      tabId="1"
      tag="div"
    >
      <Row
        tag="div"
      >
        <MealItem
          id={1}
          key="7"
          title="Food"
        />
      </Row>
    </TabPane>
    <TabPane
      tabId="2"
      tag="div"
    >
      <Row
        tag="div"
      >
        <SearchInstruction />
        <MealItem
          id={1}
          key="7"
          title="Food"
        />
      </Row>
    </TabPane>
  </TabContent>
</div>
`;
